client/adventureMap/AdventureMapInterface.cpp

```diff
@@ -153,15 +157,26 @@ void AdventureMapInterface::deactivate()
 void AdventureMapInterface::showAll(Canvas & to)
 {
 	CIntObject::showAll(to);
+	dim(to);
 	LOCPLINT->cingconsole->show(to);
 }
 
 void AdventureMapInterface::show(Canvas & to)
 {
 	CIntObject::show(to);
+	dim(to);
 	LOCPLINT->cingconsole->show(to);
 }
 
+void AdventureMapInterface::dim(Canvas & to)
+{
+	if(!GH.windows().findWindows<CCastleInterface>().empty() ||
+			!GH.windows().findWindows<CHeroWindow>().empty() ||
+			!GH.windows().findWindows<CKingdomInterface>().empty() ||
+			!GH.windows().findWindows<BattleWindow>().empty())
+		to.drawColor(Rect(0, 0, GH.screenDimensions().x, GH.screenDimensions().y), ColorRGBA(0, 0, 0, 128));
+}
+

```

Comment:
Perhaps simply dim on any window that is not popup?

Reply:
Also - possibly add option (without UI) to control it?

Reply:
For me it's also okay. I like it. But this also means for simple text windows.

Setting added...

Reply:
Good point about simple text windows...
Well, I guess let's see how it goes, if dimming is too excessive - we can add separate method to check whether window requires dimming (don't really like listing all window types)